test/index.test.ts

```diff
@@ -1300,6 +1300,63 @@ test("auth.hook(): handle 401 in first 5 seconds (#65)", async () => {
   expect(global.console.warn.mock.calls.length).toEqual(3);
 });
 
+test("auth.hook(): throw error with custom message after unsuccessful retries (#163)", async () => {
+  expect.assertions(1);
+
+  const mock = fetchMock
+    .sandbox()
+    .postOnce("https://api.github.com/app/installations/123/access_tokens", {
+      token: "secret123",
+      expires_at: "1970-01-01T01:00:00.000Z",
+      permissions: {
+        metadata: "read",
+      },
+      repository_selection: "all",
+    })
+    .get("https://api.github.com/repos/octocat/hello-world", (url) => {
+      return {
+        status: 401,
+        body: {
+          message: "Bad credentials",
+          documentation_url: "https://developer.github.com/v3",
+        },
+      };
+    });
+
+  const auth = createAppAuth({
+    id: APP_ID,
+    privateKey: PRIVATE_KEY,
+    installationId: 123,
+  });
+
+  const requestWithMock = request.defaults({
+    headers: {
+      "user-agent": "test",
+    },
+    request: {
+      fetch: mock,
+    },
+  });
+  const requestWithAuth = requestWithMock.defaults({
+    request: {
+      hook: auth.hook,
+    },
+  });
+
+  global.console.warn = jest.fn();
+
+  requestWithAuth("GET /repos/octocat/hello-world").catch((error) => {
+    expect(error.message).toBe(
+      `[@octokit/auth-app] After 3 retries within 6s after creating the installation access token, the response remains 401. At this point it's unlikely a replication lag, but a legit authentication problem or a system outage, please check https://www.githubstatus.com/`
```

Comment:
@gwestersf @imwiss This is the error message people will see when the request retries fail to account for the installation access token replication lag. Let me know if you'd like me to rephrase it in any way.

This is just the message, the error has extra properties by default with more information.

```
    RequestError [HttpError]: [@octokit/auth-app] After 3 retries within 6s after creating the installation access token, the response remains 401. At this point it's unlikely a replication lag, but a legit authentication problem or a system outage, please check https://www.githubstatus.com/
        at /Users/gregor/Projects/octokit/auth-app.js/node_modules/@octokit/request/dist-src/fetch-wrapper.js:49:31
        at sendRequestWithRetries (/Users/gregor/Projects/octokit/auth-app.js/src/hook.ts:65:12) {
      status: 401,
      headers: { 'content-length': '83', 'content-type': 'application/json' },
      request: {
        method: 'GET',
        url: 'https://api.github.com/repos/octocat/hello-world',
        headers: {
          accept: 'application/vnd.github.v3+json',
          'user-agent': 'test',
          authorization: 'token [REDACTED]'
        },
        request: { fetch: [Function], hook: [Function: bound hook] AsyncFunction }
      },
      documentation_url: 'https://developer.github.com/v3'
    }
```

Reply:
Thanks for the ping 👍  Additionally, are there any error messages returned by the API with the 401 that would make sense to surface in Octokit's error message?

Reply:
Formatting suggestion:
```
After 3 retries within 6s of creating the installation access token, the response remains 401. At this point, the cause may be an authentication problem or a system outage. Please check https://www.githubstatus.com for status information.
```

Reply:
> are there any error messages returned by the API with the 401 that would make sense to surface in Octokit's error message

No. The only other information is `documentation_url` which we add to the error object.